### ISSUE TITLE
WebSocket is not available, What This Mean ?

### ISSUE THREAD

**Issue by prakasa1904** [miguelgrinberg/Flask-SocketIO#894]
I have issue with message bellow:
```
WARNING in __init__: Flask-SocketIO is Running under Werkzeug, WebSocket is not available.
```

Every running my server, always showing error. Need help

**Comment by miguelgrinberg** [comment]
It means that the webserver that you are using cannot do websocket. If you want to enable the websocket transport you need to use a different web server. See the deployment section of the documentation to see your options.

**Comment by desmondddd** [comment]
Same problem here:

Flask==1.0.2
Flask-SocketIO==3.2.1
greenlet==0.4.15

When I want to run flask: `flask run --no-debugger --no-reload` 
with environments "FLASK_APP=app" "FLASK_ENV=production" "FLASK_DEBUG=0" 
terminal promotes: "WARNING in __init__: Flask-SocketIO is Running under Werkzeug, WebSocket is not available."

But when I changed to old Flask-SocketIO==3.1.2, everything goes well.

**Comment by miguelgrinberg** [comment]
@desmondddd this is correct. The `flask run` command cannot be used to run on the more advanced webservers anymore. This command was broken and caused problems for some people, so I'm not supporting that way of starting the server anymore, except when using the Flask dev server, which does not have WebSocket. Use `socketio.run(app)` to run your eventlet or gevent web server.

**Comment by miguelgrinberg** [comment]
As I said above, reconfigure your vsproject to run the script instead of calling `flask run`. Then add the following at the bottom of the script:

```python
if __name__ == '__main__':
    socketio.run(app, debug=True)
```

**Comment by prakasa1904** [comment]
I understand if it depend on web server supporting ws or not. But I mean, even all of requirement for supporting ws is installed but still showing that warning.

I try using uwsgi, but show this `ModuleNotFoundError: No module named 'gevent'`
anyone know why this error appears, actually that module exist on my local. Or do you have any suggest about localhost environment to support ws server :D .

**Comment by mswilson4040** [comment]
@prakasa1904 @desmondddd I am having the same issue. Is there an example repo or a little bit more guidance on running socket.io outside of the flask app?

**Comment by ishanajmeri** [comment]
> ```python
> if __name__ == '__main__':
>     socketio.run(app)
> ```

thanks for that, it's working. :)

**Comment by haliphax** [comment]
I realize this has been closed for some time, but I'm curious - is there a way to suppress this warning? I'm using Flask's CLI decorators in order to build out some database management functions, and this warning shows up each time I run one of those commands. The thing is, I'm not actually _serving_ any content while running these, so I don't care about the warning.

When I run the app as a web server, I use `socketio.run` and all is well. It's just the CLI that I want to tidy up.

**Comment by miguelgrinberg** [comment]
@haliphax the error means that there are no websocket enabled servers installed in your virtualenv. Are you running the Flask CLI on a different virtualenv than the one you use when you run the server? Normally just having eventlet or gevent-websocket installed silences this warning, even if you are running unrelated CLI commands.

**Comment by haliphax** [comment]
Nope. Same server. It's in a docker container. It may have to do with how my app is constructed; I'm not using the factory pattern.

**Comment by miguelgrinberg** [comment]
@haliphax Hmm. Okay, I think I see what the problem is. Do you want to write a separate issue? I'll look into it.

**Comment by haliphax** [comment]
Will do. Thanks!
